#! /bin/sh /usr/share/dpatch/dpatch-run
## 18_arm_xpcom_unused_attribute.dpatch by Mike Hommey <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from Steve Langasek to fix unused vs. used gcc attribute on arm.
## DP: bz#307418

@DPATCH@

=== xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp
==================================================================
--- xulrunner/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp	(revision 22)
+++ xulrunner/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp	(local)
@@ -45,7 +45,8 @@
 #endif
 
 /* Specify explicitly a symbol for this function, don't try to guess the c++ mangled symbol.  */
-static nsresult PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args) asm("_PrepareAndDispatch");
+static nsresult PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args) asm("_PrepareAndDispatch")
+__attribute__((used));
 
 static nsresult
 PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args)
